pandas/_libs/ops_dispatch.pyx

```diff
@@ -94,7 +103,11 @@ def maybe_dispatch_ufunc_to_dunder_op(
     if kwargs or ufunc.nin > 2:
         return NotImplemented
 
-    if method == "__call__" and op_name in DISPATCHED_UFUNCS:
+    if method == "__call__" and (
+        op_name in DISPATCHED_UFUNCS or (
+            _no_index and op_name in DISPATCHED_UFUNCS_NO_INDEX
```

Comment:
I initially just commented out those 3 methods in `DISPATCHED_UFUNCS`, but that would mean we will use the fallback for all objects, not just for Index. So therefore made it a bit more complex.

Passing a `_no_index` keyword in the dispatch function, to avoid having to check here whether `self` is an Index (this cython file has currently no dependencies on / imports from other files) 
And named the keyword with a leading underscore just to be sure to cause no clashes with keywords from ufuncs

Reply:
and the reason for not just patching `Index.__array_ufunc__` is?

Reply:
Because I was overthinking it?

Yes, of course, that's much simpler ;)